### PR TITLE
[WebGL] Fix NHWC packed depthwise conv2d for dilation=3

### PR DESCRIPTION
**Pull request by Linchenn**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6662)

### CODE REVIEW

File: tfjs-backend-webgl/src/conv_packed_gpu_depthwise.ts [hunk 1]
```diff
@@ -193,18 +193,21 @@ export class DepthwiseConvPacked2DProgram implements GPGPUProgram {
               if (dilationWidth > 1) {
                 mainLoop += `
                     xCOffset -= 2;
-                    if (xCOffset >= 0 && xCOffset < inDims[1] && xTexelC${
-                    colIndex}Ready == 0) {
-                      xTexelC${colIndex} = getX(batch, xR, xCOffset, d1);
-                      xTexelC${colIndex}Ready = 1;
+                    if (xCOffset >= 0 && xCOffset < inDims[1]) {
+                     previous = getX(batch, xR, xCOffset, d1);
```
Comment (Linchenn) [inline tfjs-backend-webgl/src/conv_packed_gpu_depthwise.ts:197]: In this case, it does `xCOffset -= 2` and `getX(batch, xR, xCOffset, d1)`, which means it want to read the texel right before `xTexelC${colIndex + 1}`, which is not necessarily `xTexelC${colIndex}` (for example, dilation=3).

**Review by pyu10055 — APPROVED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6662)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @lina128 and @qjia7)

**Review by qjia7 — APPROVED** [review]
LGTM, Thanks!

**Comment by lina128** [comment]
Thank you for fixing this hard bug!

**Pull request merged**